src/snowflake/snowpark/dataframe_transform_functions.py

```diff
@@ -57,6 +69,95 @@ def _validate_formatter_argument(self, fromatter):
         if not callable(fromatter):
             raise TypeError("formatter must be a callable function")
 
+    def _validate_and_extract_time_unit(
```

Comment:
This function seems like a good candidate for unit test if you have test coverage issue.

Reply:
These lines are already covered in the existing tests